File: cs25backend/src/main/java/edu/dongguk/cs25backend/config/WebMvcConfig.java
package edu.dongguk.cs25backend.config;

import edu.dongguk.cs25backend.security.UserIdArgumentResolver;
import edu.dongguk.cs25backend.security.UserIdInterceptor;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.List;

@Configuration
@RequiredArgsConstructor
public class WebMvcConfig implements WebMvcConfigurer {
    private final UserIdArgumentResolver userIdArgumentResolver;
    private final UserIdInterceptor userIdInterceptor;

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(userIdInterceptor)
                .addPathPatterns("/**")
                .excludePathPatterns("/", "/api/v1/auth/**");
    }

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> resolvers) {
        resolvers.add(userIdArgumentResolver);
    }


}

File: cs25server/src/main/kotlin/edu/dongguk/cs25server/config/WebMvcConfig.kt
package edu.dongguk.cs25server.config

import edu.dongguk.cs25server.security.UserIdArgumentResolver
import edu.dongguk.cs25server.security.UserIdInterceptor
import org.springframework.context.annotation.Configuration
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebMvcConfig (
    private val userIdArgumentResolver: UserIdArgumentResolver,
    private val userIdInterceptor: UserIdInterceptor
) : WebMvcConfigurer {

    override fun addArgumentResolvers(resolvers: MutableList<HandlerMethodArgumentResolver>) {
        resolvers.add(userIdArgumentResolver)
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(userIdInterceptor)
            .addPathPatterns("/**")
            .excludePathPatterns("/", "/api/v1/auth/**")
    }

}